### PR TITLE
Disallow Google GSON 2.10 or higher due to removed methods

### PR DESCRIPTION
**Pull request by sbluhm**
## What does this PR change?

1. Disallow Google GSON 2.10 or higher due to removed methods.
Salt communication fails otherwise with a class not found exception when accessing the Salt keys page.
(GSON 2.10 is available in the Enteprise Linux 9 repos)

2. Remove minimum GSON requirement. What would install a lower version?

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

Package build tested on Copr for  epel-9-x86_64 

- [x] **DONE**

## Links

Fixes #
Tracks # **add downstream PR, if any**

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/spacewalk-java.spec [hunk 1]
```diff
@@ -96,7 +96,7 @@ BuildRequires:  dom4j
 BuildRequires:  dwr >= 3
 BuildRequires:  glassfish-jaxb-runtime
 BuildRequires:  glassfish-jaxb-txw2
-BuildRequires:  google-gson >= 2.2.4
+BuildRequires:  google-gson < 2.10.0
```
Comment (deneb-alpha) [inline java/spacewalk-java.spec:99]: What about something like:
```suggestion
BuildRequires:  (google-gson >= 2.2.4 with google-gson < 2.10.0)
```

File: java/spacewalk-java.spec [hunk 2]
```diff
@@ -186,7 +186,7 @@ Requires:       glassfish-activation-api
 Requires:       glassfish-jaxb-api
 Requires:       glassfish-jaxb-runtime
 Requires:       glassfish-jaxb-txw2
-Requires:       google-gson >= 2.2.4
+Requires:       google-gson < 2.10.0
```
Comment (deneb-alpha) [inline java/spacewalk-java.spec:189]: Like for the Buildrequires,  could this also work for you?
```suggestion
Requires:       (google-gson >= 2.2.4 with google-gson < 2.10.0)
```

**Review by deneb-alpha — CHANGES_REQUESTED** [review]
I left a comment Stefan, what's your take?

File: java/spacewalk-java.spec (hunk 1 above)
Comment (sbluhm) [reply java/spacewalk-java.spec:99]: Let me try your suggestion. I tried getting both requirement statements to work but didn't succeed.

**Comment by sbluhm** [comment]
Thanks a lot @deneb-alpha , your suggestion works fine!

**Comment by deneb-alpha** [comment]
notes:
* failure on: `acceptance-tests-secondary-parallel / test-uyuni (2) / test-uyuni (pull_request) ` is unrelated
* failure on: `acceptance-tests-secondary-parallel / test-uyuni (5) / test-uyuni (pull_request)` is unrelated

**Review by deneb-alpha — APPROVED** [review]
LGTM, thanks

**Pull request merged**